scaleway/resource_compute_instance_server.go

```diff
@@ -431,7 +431,9 @@ func resourceScalewayComputeInstanceServerUpdate(d *schema.ResourceData, m inter
 		}
 	}
 
-	updateRequest.Volumes = &volumes
+	if len(volumes) > 0 {
```

Comment:
Should you move it inside `if raw, ok := d.GetOk("additional_volume_ids"); d.HasChange("additional_volume_ids") && ok ` 

This way we do not need this if